File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char,int> mp;
        mp['I'] = 1,mp['V'] = 5,mp['X'] = 10,mp['L'] = 50,mp['C'] = 100,mp['D'] = 500,mp['M'] = 1000;

        int n = s.size();
        int sum = 0;
        for(int i=0;i<n;i++){
            if(i!=n-1 and mp[s[i]] < mp[s[i+1]]) sum -= mp[s[i]];
            else sum += mp[s[i]];
        }
        return sum;

    }
};

File: 13-roman-to-integer/roman-to-integer.java
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> hm = new HashMap<>();
        hm.put('I', 1);
        hm.put('V', 5);
        hm.put('X', 10);
        hm.put('L', 50);
        hm.put('C', 100);
        hm.put('D', 500);
        hm.put('M', 1000);

        int n = s.length();
        int sum = 0;
        
        for (int i = 0; i < n; i++) {
            int currentVal = hm.get(s.charAt(i));
            
            if (i < n - 1 && currentVal < hm.get(s.charAt(i + 1))) {
                sum -= currentVal;
            } else {
                sum += currentVal;
            }
        }
        
        return sum;
    }
}